From 324493e716a2e5fa60b6b013d5df831b03f2a678 Mon Sep 17 00:00:00 2001
From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>
Date: Mon, 1 Oct 2018 10:54:48 +0100
Subject: migration/postcopy: Clear have_listen_thread

RH-Author: Dr. David Alan Gilbert <dgilbert@redhat.com>
Message-id: <20181001105449.41090-2-dgilbert@redhat.com>
Patchwork-id: 82326
O-Subject: [RHEL-8.0 qemu-kvm PATCH 1/2] migration/postcopy: Clear have_listen_thread
Bugzilla: 1608765
RH-Acked-by: Pankaj Gupta <pagupta@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Laurent Vivier <lvivier@redhat.com>

From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>

Clear have_listen_thread when we exit the thread.
The fallout from this was that various things thought there was
an ongoing postcopy after the postcopy had finished.

The case that failed was postcopy->savevm->loadvm.

This corresponds to RH bug https://bugzilla.redhat.com/show_bug.cgi?id=1608765

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
Message-Id: <20180914170430.54271-2-dgilbert@redhat.com>
Reviewed-by: Peter Xu <peterx@redhat.com>
Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
(cherry picked from commit 9cf4bb8730c669c40550e635a9e2b8ee4f1664ca)
  Manual merge due to context

Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 migration/savevm.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/migration/savevm.c b/migration/savevm.c
index 7f92567..762c4b2 100644
--- a/migration/savevm.c
+++ b/migration/savevm.c
@@ -1676,6 +1676,7 @@ static void *postcopy_ram_listen_thread(void *opaque)
     migration_incoming_state_destroy();
     qemu_loadvm_state_cleanup();
 
+    mis->have_listen_thread = false;
     return NULL;
 }
 
-- 
1.8.3.1

